### PR TITLE
Enable parser profiling features

### LINKED ISSUE #356: Profiling unsupported
Now that I have my vscode debugger almost ready I wanted to add profiling support for grammars, only to get an exception that this is not supported. What are your plans wrt to this? What's missing for profiling a grammar?

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #356 

I used the same strategy to fix the other circular dependency problem in **Parser.ts**, so now `compileParseTreePattern` should work in this target as well.

### CODE REVIEW

File: src/Recognizer.ts [hunk 1]
```diff
@@ -158,8 +158,8 @@ export abstract class Recognizer<TSymbol, ATNInterpreter extends ATNSimulator> {
 	 *
 	 * @since 4.3
 	 */
-	get parseInfo(): ParseInfo | undefined {
-		return undefined;
+	get parseInfo(): Promise<ParseInfo | undefined> {
```
Comment (BurtHarris) [inline src/Recognizer.ts]: Might be clearer if we use async functions instead of explicit promises.

File: src/Parser.ts [hunk 2]
```diff
@@ -854,29 +853,33 @@ export abstract class Parser extends Recognizer<Token, ParserATNSimulator> {
 	}
 
 	@Override
-	get parseInfo(): ParseInfo | undefined {
-		let interp: ParserATNSimulator = this.interpreter;
-		if (interp instanceof ProfilingATNSimulator) {
-			return new ParseInfo(interp);
-		}
+	get parseInfo(): Promise<ParseInfo | undefined> {
```
Comment (BurtHarris) [inline src/Parser.ts]: Might be clearer if we use async function rather than promises.

File: src/Recognizer.ts (hunk 1 above)
Comment (BurtHarris) [reply src/Recognizer.ts]: I also think it might be a good convention to use an Async suffix on method names if they return promises.

**Comment by sharwell** [comment]
> Have you considered using async functions rather than explicit promises?

👍 Will try this. Hopefully cleans up nicely.

**Comment by BurtHarris** [comment]
I did some research on adding the Async suffix.   It seems some TypeScript frameworks like angular have decided against it.   See https://github.com/johnpapa/angular-styleguide/issues/396.

**Pull request merged**